dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/FixedSizeClassPathRepository.java

```diff
@@ -48,18 +48,23 @@
   private final Cache<String, JavaClass> loadedClass;
 
   /**
-   * Mapping from class names to special class file locations.
+   * Mapping from class names to file names.
    *
-   * <p>While class name and class file name are the same in most cases, sometimes classes are not
-   * placed in the root of a JAR file to support a framework-specific JAR structure. For example,
-   * Spring Boot Gradle Java plugin places class files under "BOOT-INF/classes". To load such
-   * classes by class name, this mapping keeps track of the special location once they are loaded.
+   * <p>Class names are the fully package qualified names found in Java source code such as
+   * {@code com.google.Foo}. The byte code of this class is normally found in a JAR at the path
+   * com/google/Foo.class. In this case {@code com.google.Foo} is also the file name.
+   * 
+   * A few tools relocate classes into different directories within the JAR and use a special

```

Comment:
```suggestion
   * <p>A few tools relocate classes into different directories within the JAR and use a special
```

I like the clear explanation of the difference.